### PR TITLE
Add password expiry hint in user detail page

### PR DESCRIPTION
**Pull request by IniZio**
ref DEV-1625

### Screens

Case|Screen
-|-
Normal|Screenshot 2024-07-29 at 12-24-09 Authgear Portal
Manual expired|Screenshot 2024-07-29 at 12-24-33 Authgear Portal
Last update expired|Screenshot 2024-07-29 at 12-23-38 Authgear Portal

### CODE REVIEW

**Review by louischan-oursky — COMMENTED** [review]
SCR-20240730-lizh

---
SCR-20240730-lkkr

**Comment by jimcky** [comment]
@fungc-io we were also asking on Figma, but since there are annotated screenshots, it might be easier for you to read here.

Can you comment on the design questions above and let us know what you think?

**Comment by fungc-io** [comment]
@louischan-oursky @jimcky @IniZio 
> - The text is confusing...
  - how about removing the text below and change the label above to "Ask to change password on login"
image

> - expiry hint

How about changing the text to: `The password is expired for 32 days. The user will be asked to update it on their next login. `

**Comment by jimcky** [comment]
Sounds good to me - @IniZio ok @_@?

**Comment by louischan-oursky** [comment]
Looks good to me.

**Comment by IniZio** [comment]
Updated the toggle label

image

File: authui/Makefile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+.PHONY: check-tidy
+check-tidy:
+	make fmt
+	git status --porcelain | grep '.*'; test $$? -eq 1
+
+.PHONY: fmt
+fmt:
+	yarn eslint:format
```
[the remark this answers is not in the document]
Comment (IniZio) [reply authui/Makefile]: Changed to npm run

File: scripts/sh/check-dockerignore.sh [hunk 2]
```diff
@@ -13,6 +13,9 @@ EOF
 # This command outputs a sorted list of files in the build context.
 docker run --rm build-context | sort > /tmp/authgear-dockerignore
 
+# Filter out files listed in .git/info/exclude
+grep -v -f .git/info/exclude /tmp/authgear-dockerignore > /tmp/authgear-dockerignore
+
```
Comment (louischan-oursky) [inline scripts/sh/check-dockerignore.sh:18]: Remove this line.

File: .gitignore [hunk 3]
```diff
@@ -19,6 +19,8 @@ docker-compose.override.yaml
```
Comment (louischan-oursky) [inline .gitignore:19]:
```
# .editorconfig
.editorconfig
```

Comment (IniZio) [reply .gitignore:19]: Updated and reverted dockerignore script

File: Makefile [hunk 4]
```diff
@@ -66,11 +66,17 @@ lint:
 	git diff --exit-code .make-lint-expect > /dev/null 2>&1
 	go run ./devtools/gotemplatelinter ./resources/authgear/templates/en/web/authflowv2
 
+	make -C authui lint
+	make -C portal lint
+
```
Comment (louischan-oursky) [inline Makefile:71]: Remove these.

File: Makefile [hunk 5]
```diff
@@ -66,11 +66,17 @@ lint:
...
 .PHONY: fmt
 fmt:
 	# Ignore generated files, such as wire_gen.go and *_mock_test.go
 	find ./pkg ./cmd ./e2e -name '*.go' -not -name 'wire_gen.go' -not -name '*_mock_test.go' | sort | xargs goimports -w -format-only -local github.com/authgear/authgear-server
 
+	make -C authui fmt
+	make -C portal fmt
```
Comment (louischan-oursky) [inline Makefile:78]: Remove these.

File: Makefile [hunk 6]
```diff
@@ -108,6 +114,9 @@ check-tidy:
 	# (cd custombuild && go mod tidy)
 	git status --porcelain | grep '.*'; test $$? -eq 1
 
+	make -C authui check-tidy
+	make -C portal check-tidy
+
```
Comment (louischan-oursky) [inline Makefile:119]: Remove these.

File: authui/Makefile [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+	npm run eslint:format
+	npm run stylelint:format
+	npm run prettier:format
+
+.PHONY: lint
+lint:
+	npm run eslint
+	npm run stylelint
+	npm run prettier
+	npm run typecheck
```
Comment (louischan-oursky) [inline authui/Makefile:17]: typecheck
eslint
stylelint
prettier

File: .github/workflows/ci.yaml [hunk 8]
```diff
@@ -65,7 +65,7 @@ jobs:
     - run: npm run stylelint
       working-directory: ./portal
       if: ${{ !cancelled() }}
-    - run: npm run fmt
+    - run: npm run prettier
```
Comment (louischan-oursky) [inline .github/workflows/ci.yaml:68]: In authui_test, add `run: make check-tidy`

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**